### DISCUSSION TITLE
Auth0 Client set cookie same site to prevent CSRF attack

### DISCUSSION THREAD

**Discussion by fabio-percivaldi** [mia-platform/community#78] [Q&A]
in order to avoid CSRF attack it is suggested to implement a same-site cookie https://portswigger.net/web-security/csrf/samesite-cookies

this suggestion has been given by Intrè penetration tester. 

what do you think of implementing this in the /oauth/token API? we should return this cookie togheter with the sid

**Comment by fredmaggiowski** [comment]
Hi @fabio-percivaldi this same request has been done previously by another collegue; i asked him to implement it, i guess we could increase the priority for this.

**Reply by fredmaggiowski** [reply]
@albertotessarotto let's talk with @ste23droid and @fabio-percivaldi to understand the priority for this feature then we can understand together who actually implements it. This is a small improvements that have important security implications, let's not forget about it.

**Comment by davidebianchi** [comment]
Hi @fabio-percivaldi! By default now browsers [set cookie to Lax](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#lax). 

Yes, we should add an enforcement on cookie and set the same-site opts!

**Comment by ste23droid** [comment]
A possible solution may be to add an additional auth0client configuration specifying the desired sameSite value.
The default may remain "no enforment" as it is now or Lax. 
For example 

` "clients": {
    "myClient1": {
          ....
         "sameSite": "Strict"
    },
   "myClient2": {
            ....
          "sameSite": "Lax"
      }
}`

**Reply by fredmaggiowski** [reply]
yes this is what we had discussed earlier; I think that this can be featured in a PR if you can open it that would be great!

**Comment by peppemanzi** [comment]
Since the discussion is inactive I close it, after checking that the issue was actually tracked internally.
